server/pulp/plugins/file/distributor.py

```diff
@@ -212,9 +213,13 @@ def _symlink_unit(self, build_dir, unit, target_paths):
         :param target_paths: The list of paths the unit should be symlinked to.
         :type  target_paths: list of L{str}
         """
+        cwd = os.path.abspath(os.path.curdir)
```

Comment:
This looks just wrong: Why is all this path computation done with `cwd` and then, the resulting path is appended to `build_dir`?

Anyway, both `cwd` and `build_dir` are implementation details (and may depend on configuration, I don't know). The validation of `target_path` should not depend on the location of `cwd` nor `build_dir`.

Reply:
The reason is that `abspath`, when e.g expanding `..`, calculates based on `cwd` rather than the `build_dir` thus the prefix calculation has to follow suit or one has to `$ cd build_dir` first; I chose the former.
The semantics of an absolute `target_path` here is as if it was a `chroot` into the repository (`build_dir`):
```
In [22]: build_dir = '/var/lib/pulp/build_dir/my_repo'

In [23]: cd ~
/home/vagrant

In [24]: cwd = os.path.abspath(os.path.curdir)

In [25]: cwd
Out[25]: '/home/vagrant'

In [26]: target_path = '/foo/bar/baz'

In [27]: abs_target_path = os.path.abspath(target_path)

In [28]: target_path_cwd_prefix = os.path.commonprefix([cwd, os.path.dirname(abs_target_path)])

In [29]: confined_target_path = abs_target_path.replace(target_path_cwd_prefix, '', 1).lstrip('/')

In [30]: os.path.join(build_dir, confined_target_path)
Out[30]: '/var/lib/pulp/build_dir/my_repo/foo/bar/baz'
```